contracts/cw3-fixed-multisig/src/contract.rs

```diff
@@ -383,22 +396,61 @@ fn list_voters(
     limit: Option<u32>,
 ) -> StdResult<VoterListResponse> {
     let limit = limit.unwrap_or(DEFAULT_LIMIT).min(MAX_LIMIT) as usize;
-    let start = start_after.map(|s| Bound::ExclusiveRaw(s.into()));
-
-    let voters = VOTERS
-        .range(deps.storage, start, None, Order::Ascending)
-        .take(limit)
-        .map(|item| {
-            item.map(|(addr, weight)| VoterDetail {
-                addr: addr.into(),
-                weight,
-            })
-        })
-        .collect::<StdResult<_>>()?;
+    let start_address = start_after.map(|addr| Addr::unchecked(addr));
+
+    let addresses = match start_address {
+        Some(addr) => VOTER_ADDRESSES.iter_from(deps.storage, &addr)?,
+        None => VOTER_ADDRESSES.iter(deps.storage),
+    };
+
+    let mut voters = vec![];
+    let mut ctr = 0;
+    for addr in addresses {
+        let weight = match VOTERS.get(deps.storage, &addr) {
+            Some(weight) => weight,
+            None => continue,
+        };
```

Comment:
would a voter not have a weight?

Reply:
You're correct in pointing out that the `None` arm will likely never happen. If it does, something must have gone seriously wrong.

Reply:
maybe do this then, that way contract owners can see if its messed up
```suggestion
        let weight = VOTERS.get(deps.storage, &addr).unwrap_or_default();
```